### PR TITLE
improvement(packages): change .sphere.io .commercetools.co to .europe-west1.gcp.commercetools.com

### PR DESCRIPTION
**Pull request by MarkusAzer**
#### Summary

Following Issue [1482](https://github.com/commercetools/nodejs/issues/1482) , I replaced sphere.io, .commercetools.co To .europe-west1.gcp.commercetools.com

### CODE REVIEW

File: docs/cli/category-exporter.md [hunk 1]
```diff
@@ -59,7 +59,7 @@ import fs from 'fs'
 
 const options = {
     apiConfig: {
-      apiUrl: 'https://api.sphere.io'
+      apiUrl: 'https://api.europe-west1.gcp.commercetools.com'
       host: 'https://auth.sphere.com'
```
Comment (emmenko) [inline docs/cli/category-exporter.md]: What about this one?

File: docs/cli/resource-deleter.md [hunk 2]
```diff
@@ -50,9 +50,9 @@ Options:
   --version                  Show version number.                       [boolean]
   --output, -o               Path to output file.            [default: "stdout"]
   --apiUrl                   The host URL of the HTTP API service.
-                                              [default: "https://api.sphere.io"]
+                                              [default: "https://api.europe-west1.gcp.commercetools.com"]
   --authUrl                  The host URL of the OAuth API service.
-                                             [default: "https://auth.sphere.io"]
+                                             [default: "https://https://docs.commercetools.com/http-api-authorization#http-api---authorization"]
```
Comment (emmenko) [inline docs/cli/resource-deleter.md]: Those urls are wrong

File: docs/cli/category-exporter.md (hunk 1 above)
Comment (MarkusAzer) [reply docs/cli/category-exporter.md]: prefect and thanks :), i really didnt notice it

**Comment by emmenko** [comment]
What about my point?

> From my point of view, we should also remove the default URL values and just require them.

/cc @daern91 @katmatt

**Comment by katmatt** [comment]
@emmenko You mean that we require that you have to pass in the urls? That would be a breaking change right? Which is fine with me, but should then be reflected in our version update.

**Comment by emmenko** [comment]
Yes of course. It would be a breaking change. There should be no problem with that from my side.

**Comment by katmatt** [comment]
@emmenko I agree with you that with our multi cloud setup it doesn't make a lot of sense to provide defaults. I personally like the idea to have an enum that lists all of our environments and you already suggested it for [labd/terraform-provider-commercetools](https://github.com/labd/terraform-provider-commercetools) . The enums would then provide the auth and api url and that would improve our developer experience. 
That's why I would suggest to keep the default urls and instead discuss how to implement the enum solution. Why should we waste time by fixing a feature that we want to improve?

**Comment by emmenko** [comment]
You mean that we only require `region` and `cloudRegion` and the SDK will then build the proper URLs internally?

Yeah we could also do that but we should be consistent will all other SDKs.
For now I would suggest to just drop the default, and start a discussion with the other SDK teams to look into the more generic approach.

WDYT?

**Comment by katmatt** [comment]
@emmenko all of our other SDKs kept the default urls :wink:  And in our generated Javs SDKs we stopped providing default urls.

/CC @jenschude

**Comment by jenschude** [comment]
I would go a two step approach. First replace the default values and release a minor version and afterwards remove the defaults and require the URLs by releasing a new major version. So it would still be possible to upgrade to the new URLs without touching the code base.

About the enums I had the discussion also with Nikolaus once. The issue is that we have to parameters in the Platform API (importer too?) and for example only one for the ML services and the two parameters can't be used in all combinations.

This would have led to introduce an enum with the Full-URI and a meaning full name. Btw in RAML it's not possible to define parameters in the AuthURIs 🤦‍♂

**Comment by daern91** [comment]
I was thinking of a two-step approach too and think it makes the most sense.

**Comment by katmatt** [comment]
This PR is already quite big and dropping support for default urls seems to require some more discussions:

- Do we drop support for default urls in all of our cli applications too?
- Do we want to implement an enum based better solution or do we still want to have separate auth and api urls?

That's why I would suggest to merge this PR, which already fixed what the associated issue #1482 describes. And we then should discuss the proper solution separately.

**Comment by emmenko** [comment]
Yep, we can of course tackle the other points in other PRs.

File: packages/sdk-middleware-auth/test/anonymous-session-flow.spec.js [hunk 3]
```diff
@@ -16,7 +16,7 @@ function createTestRequest(options) {
 
 function createTestMiddlewareOptions(options) {
   return {
-    host: 'https://auth.commercetools.co',
+    host: 'https://auth.europe-west1.gcp.commercetools.com',
```
Comment (daern91) [inline packages/sdk-middleware-auth/test/anonymous-session-flow.spec.js]: just as a heads up, technically these pointed towards US env but shouldn't matter for tests.

**Pull request merged**